lib/ocpp/v201/connectivity_manager.cpp

```diff
@@ -121,6 +142,50 @@ bool ConnectivityManager::send_to_websocket(const std::string& message) {
     return this->websocket->send(message);
 }
 
+void ConnectivityManager::on_network_disconnected(const std::optional<int32_t> configuration_slot,
+                                                  const std::optional<OCPPInterfaceEnum> ocpp_interface) {
+
+    if (!configuration_slot.has_value() and !ocpp_interface.has_value()) {
+        EVLOG_warning << "Network disconnected. Not clear which network is disconnected: configuration slot and ocpp "
+                         "interface are empty";
+        return;
+    }
+
+    const int actual_configuration_slot = get_active_network_configuration_slot();
+    std::optional<NetworkConnectionProfile> network_connection_profile =
+        this->get_network_connection_profile(actual_configuration_slot);
+
+    if (!network_connection_profile.has_value()) {
+        EVLOG_warning << "Network disconnected. No network connection profile configured";
+    } else if ((configuration_slot.has_value() and (configuration_slot.value() == actual_configuration_slot)) or
+               (ocpp_interface.has_value() and
+                (ocpp_interface.value() == network_connection_profile.value().ocppInterface))) {
+        this->disconnect_websocket(ocpp::WebsocketCloseReason::GoingAway);
+        // Since there is no connection anymore: reconnect with the next available network connection profile.
+        reconnect(ocpp::WebsocketCloseReason::GoingAway);
```

Comment:
After the PR https://github.com/EVerest/libocpp/pull/827 the reason is sent to the callback register_closed_callback. Why don't we take advantage of here and call reconnect from one place? Maybe do we need to think about the new reason for disconnect?

Reply:
I'm not entirely sure what you mean or would like me to change here. Could you elaborate a bit more?

Reply:
You can call `reconnect` from the one place (from the `register_closed_callback`), that gives better readability and maintainability. I would use this binding: call `disconnect_websocket` with a reason and in the callback `register_closed_callback` depending on the reason (restart or switch to another network profile), that is now propagated there, call `reconnect`